src/contexts/WalletProvider.tsx

```diff
@@ -175,7 +176,11 @@ export function WalletContextProvider({ children }: { children: any }) {
     [request],
   );
 
-  if (!isWalletLoading && isWalletLocked) {
+  // We do not require extension to be unlocked for wallet selection
+  const routeMatch = useRouteMatch('/approve/select-wallet');
```

Comment:
Maybe it would be better to have a separate provider for locked/unlocked state (like a `LockProvider`), but I wasn't sure if it's worth it since it's just one route.

Reply:
Let's keep it like this for now. We can refactor when we have more